nnunetv2/training/dataloading/nnunet_dataset.py

```diff
@@ -110,6 +112,220 @@ def load_case(self, key):
+    Also this is still in the nnunet library so it is ok. 
+    '''     
+    def __init__(self, folder: str, patch_size: Union[List[int], Tuple[int, ...], np.ndarray],
+                 final_patch_size: Union[List[int], Tuple[int, ...], np.ndarray],
+                 label_manager: LabelManager,                 
+                 case_identifiers: List[str] = None,                 
+                 oversample_foreground_percent: float = 0.0,
+                 num_images_properties_loading_threshold: int = 0,
+                 folder_with_segs_from_previous_stage: str = None):
+        # Initialize 
+        nnUNetDataset.__init__(self, folder, case_identifiers, num_images_properties_loading_threshold,
+                 folder_with_segs_from_previous_stage)
+        Dataset.__init__(self)
+        self.final_patch_size = final_patch_size
+        self.patch_size = patch_size
+        self.need_to_pad = (np.array(patch_size) - np.array(final_patch_size)).astype(int)
+        self.has_ignore = label_manager.has_ignore_label
+        self.annotated_classes_key = tuple(label_manager.all_labels)
+        self.oversample_foreground_percent = oversample_foreground_percent
+    
+    def _get_case_dict(self, key):
+        ret = {**self.dataset[key]}
+        if 'properties' not in ret.keys():
+            ret['properties'] = load_pickle(ret['properties_file'])
+        return ret
+    
+    def load_case(self, idx):
+        # This is different because Pytorch indexes with numbers not keys
+        key = list(self.keys())[idx]
+        entry = self._get_case_dict(key)
+        if 'open_data_file' in entry.keys():
+            data = entry['open_data_file']
+            # print('using open data file')
+        elif isfile(entry['data_file'][:-4] + ".npy"):
+            data = np.load(entry['data_file'][:-4] + ".npy", 'r')
+            if self.keep_files_open:
+                self.dataset[key]['open_data_file'] = data
+                # print('saving open data file')
+        else:
+            data = np.load(entry['data_file'])['data']
+
+        if 'open_seg_file' in entry.keys():
+            seg = entry['open_seg_file']
+            # print('using open data file')
+        elif isfile(entry['data_file'][:-4] + "_seg.npy"):
+            seg = np.load(entry['data_file'][:-4] + "_seg.npy", 'r')
+            if self.keep_files_open:
+                self.dataset[key]['open_seg_file'] = seg
+                # print('saving open seg file')
+        else:
+            seg = np.load(entry['data_file'])['seg']
+
+        if 'seg_from_prev_stage_file' in entry.keys():
+            if isfile(entry['seg_from_prev_stage_file'][:-4] + ".npy"):
+                seg_prev = np.load(entry['seg_from_prev_stage_file'][:-4] + ".npy", 'r')
+            else:
+                seg_prev = np.load(entry['seg_from_prev_stage_file'])['seg']
+            seg = np.vstack((seg, seg_prev[None]))
+
+        return data, seg, entry['properties']
```

Comment:
In the existing implementation should be something like
```python
def load_case(self, index: int) -> Tuple[np.ndarray, np.ndarray, PropertiesDict]:
    key = list(self.keys())[idx]
    return super().load_case(key)
```

With the updated way:
```python
def load_case(self, index: int) -> Tuple[np.ndarray, np.ndarray, PropertiesDict]:
    key = list(self.source_nnunet_dataset.keys())[idx]
    return self.source_nnunet_dataset.load_case(key)
```